src/oscar/apps/checkout/views.py

```diff
@@ -659,6 +660,12 @@ class ThankYouView(generic.DetailView):
     template_name = 'oscar/checkout/thank_you.html'
     context_object_name = 'order'
 
+    def get(self, request, *args, **kwargs):
+        obj = self.get_object()
```

Comment:
Could I suggest we just replace the super method entirely, to avoid the queries in `get_object()` running twice? So just:

```
def get(self, request, *args, **kwargs):
    self.object = self.get_object()
    if self.object is None:
        return redirect(settings.OSCAR_HOMEPAGE)
    context = self.get_context_data(object=self.object)
    return self.render_to_response(context)
```

Reply:
Yes, makes sense.

Reply:
Alright done.